addons/mrp/models/mrp_bom.py

```diff
@@ -323,21 +324,19 @@ def _compute_operation_count(self):
         for bom in self:
             bom.operation_count = len(bom.operation_ids)
 
-    def action_compute_bom_days(self):
-        company_id = self.env.context.get('default_company_id', self.env.company.id)
-        warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
+    @api.depends('bom_line_ids', 'product_id', 'company_id')
+    def _compute_estimated_days(self):
         for bom in self:
-            bom_data = self.env['report.mrp.report_bom_structure'].with_context(minimized=True)._get_bom_data(bom, warehouse, bom.product_id, ignore_stock=True)
-            bom.days_to_prepare_mo = self.env['report.mrp.report_bom_structure']._get_max_component_delay(bom_data['components'])
-            if bom_data.get('availability_state') == 'unavailable' and not bom_data.get('components_available', True):
-                return {
-                    'type': 'ir.actions.client',
-                    'tag': 'display_notification',
-                    'params': {
-                        'title': _('Cannot compute days to prepare due to missing route info for at least 1 component or for the final product.'),
-                        'sticky': False,
-                    }
-                }
+            company_id = bom.env.context.get('default_company_id', bom.env.company.id)
+            warehouse = bom.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
+            bom_data = bom.env['report.mrp.report_bom_structure'].with_context(minimized=True)._get_bom_data(bom, warehouse, bom.product_id, ignore_stock=True)
+            component_info = max(((component.get('availability_delay', 0), component.get('name')) for component in bom_data['components'] if
+                                component.get('availability_delay', 0) is not False), default=(0, ""))
+            max_delay, component_name = component_info
+            if max_delay and component_name:
```

Comment:
```suggestion
            if max_delay:
```
Isn't it sufficient?